test/decl/protocol/req/associated_type_inference.swift

```diff
@@ -588,3 +588,13 @@ extension SR_12707_P2 {
 struct SR_12707_Conform_P2: SR_12707_P2 {
   typealias A = Never
 }
+
+// SR-13172: Inference when witness is an enum case
+protocol SR_13172_P {
+  associatedtype Bar
+  static func bar(_ value: Bar) -> Self
```

Comment:
Can you add a test for inferring an associated type from a witness that is an enum case without a payload, as well?

Reply:
Sure, I have done that now.